src/main/java/com/clairvoyant/weather/controller/WeatherController.java

```diff
@@ -0,0 +1,95 @@
+package com.clairvoyant.weather.controller;
+
+import static com.clairvoyant.weather.contants.WeatherConstants.WEATHER_END_POINT;
+
+import com.clairvoyant.weather.dto.WeatherDto;
+import com.clairvoyant.weather.exception.NotFoundException;
+import com.clairvoyant.weather.service.WeatherService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * @author Gufran Khan
+ * @version 1.0
+ * @date 17-08-2021 16:19
+ */
+
+@RestController
+@Slf4j
+public class WeatherController {
```

Comment:
If we are using RouterFunction, we can delete this controller.

Reply:
here I have use different api for functional and Controller based, so that I future I can relate them(for better understanding), that's way I have not deleted. Should I delete ?

Reply:
got your point, I will remove this